### PR TITLE
CI rustdoc generation

### PR DESCRIPTION
**Pull request by ptagl**
This PR introduces a new GitHub Actions workflow to automatically generate the Rust documentation through `cargo doc` and then push it to GitHub Pages.

For simplified navigation, the documentation homepage is configured to redirect to the `nh_node` crate homepage.

Additionally, the workflow used for the Docker image build and push is now bundled into a "tag" workflow that includes the documentation as well.

### CODE REVIEW

**Review by a-petrini — APPROVED** [review]
Looks good, thanks for this PR!

File: .github/workflows/CI-rustdoc.yml [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        uses: actions/cache@v4
+        continue-on-error: false
+        with:
+          path: |
+            ~/.cargo/bin/
+            ~/.cargo/registry/index/
+            ~/.cargo/registry/cache/
+            ~/.cargo/git/db/
+            target/
+          key: ${{ runner.os }}-lint-format-cargo-${{ hashFiles('**/Cargo.lock') }}
```
Comment (JackPiri) [inline .github/workflows/CI-rustdoc.yml:29]: Is `lint-format` cache here used on purpose? My feeling would be to use `build-test` cache, but if your is a motivated choice just ignore this comment.

**Review by JackPiri — APPROVED** [review]
The PR is fine to me.
Thanks

File: .github/workflows/CI-rustdoc.yml (hunk 1 above)
Comment (ptagl) [reply .github/workflows/CI-rustdoc.yml]: It makes sense to me, thanks! I've also updated the `cargo doc` command adding the `--release` flag as the cache is built with the same flag.

**Pull request merged**